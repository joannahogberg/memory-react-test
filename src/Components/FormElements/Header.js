import React from "react";
import Button from "./Button.js";
import Row from "./Row.js";

function Header(props) {
  const userStats =
    props.gamesPlayed > 0 ? (
      <Row>
        <ul className="list-reset mt-4 leading-normal">
          <li>Total score: {props.cachedPoints}</li>
          <li>GamesPlayed: {props.gamesPlayed}</li>
          <li>Avarage points: {props.avaragePoints}</li>
        </ul>  
      </Row>
    ) : (
      <Row>
        <h4 className="text-grey-darker m-2">You have no saved statics</h4>
      </Row>
    );

  return (
    <header className="flex flex-wrap">
      <Row>
        <h1 className="text-center font-black">Memory Game</h1>
      </Row>
      {userStats}
      {props.gamesPlayed > 0 && (
        <Row>
          <Button
            onClick={props.onClick}
            className="bg-transparent hover:bg-red-light text-red-light font-semibold hover:text-white py-2 px-4 border border-red hover:border-transparent rounded"
            title="CLEAR STATS"
          />
        </Row>
      )}
    </header>
  );
}

export default Header;
