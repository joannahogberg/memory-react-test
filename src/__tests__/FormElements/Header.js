import React from 'react';
import toJSON from 'enzyme-to-json';
import { shallow, mount } from 'enzyme';
import Header from '../../Components/FormElements/Header.js';

it("Check snapshot for header", () => {
    const wrapper = shallow(<Header />);
    expect(toJSON(wrapper))
    .toMatchSnapshot("Header snap");
});

it("Look for h1", () => {
    const wrapper = shallow(<Header />);
    const h1 = wrapper.find('h1')
    expect(h1.text()).toContain('Memory Game')
    expect(h1.props().className).toContain('text-center')
});

it("Check that button and list is rendered when lgamesPlayed > 0", () => {
    const onClick = jest.fn();
    const games = 2;
    const wrapper = mount(<Header gamesPlayed={games} onClick={onClick}/>);
    const btn = wrapper.find('button')
    const list = wrapper.find('ul').children();
    expect(btn.text()).toEqual('CLEAR STATS')
    expect(list).toHaveLength(3)
    btn.simulate('click')
    expect(onClick).toHaveBeenCalled()   
});
