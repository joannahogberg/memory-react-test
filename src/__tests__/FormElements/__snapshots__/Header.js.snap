// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snap 1`] = `
<header
  className="flex flex-wrap"
>
  <Row>
    <h1
      className="text-center font-black"
    >
      Memory Game
    </h1>
  </Row>
  <Row>
    <h4
      className="text-grey-darker m-2"
    >
      You have no saved statics
    </h4>
  </Row>
</header>
`;
